qiskit/circuit/classical/types/types.py

```diff
@@ -115,3 +111,21 @@ def __hash__(self):
 
     def __eq__(self, other):
         return isinstance(other, Uint) and self.width == other.width
+
+
+@typing.final
+class Float(Type, metaclass=_Singleton):
+    """A floating point number of unspecified width.
+    In the future, this may also be used to represent a fixed-width float.
+    """
```

Comment:
In retrospect: I think we can say that this produces a double-precision IEE-754 by default, since that's what we'll actually be storing.  Nicer to be explicit, I think.

Any nastiness with quads, singles or halves can be handled some other time, but defaulting to doubles is fine, I think.

Reply:
Should we also emit a `float[64]` in QASM, though? I guess this would happen if a user were to create a `Cast` expression to a `Float`, e.g. QASM would emit `(Float[64])my_int` I think.

Reply:
Yeah, I'm saying let's treat default-constructed `Float()` in Qiskit land as if it were `float[64]` in OpenQASM land?  If we later want to specify some `width=None` override in Qiskit land, we could make `width=64` the default.

I'm mostly thinking that the constants available to users in Python land _are_ `float[64]`, so we may as well just represent that reality verbatim.